### DISCUSSION TITLE
Moved Build to Alpine | Add data.tar.gx | And other hotfixes (#213)

### DISCUSSION THREAD

**Discussion by austinsonger** [Secure-Compliance-Solutions-LLC/GVM-Docker#222] [General]
- Use the latest Alpine Version under Major Version 3 like 3.14 as of the current date. 
- Fix the "mv: inter-device move failed" error which can occur under certain circumstances.
- Add a initial data.tar.gz - no generator yet created - will follow by separate PR
- No Symlink-fu keep the setup as simple as possible
- supervisord support
  - fix #206  
- postfix support

**We have two build ways:**
> With build for initialize on container start (only 1100MB)

`docker build -f Dockerfile -t securecompliance/gvm:v21.04-latest-nodata -t securecompliance/gvm:latest  .`

or

> With pre-initialized that can be used without extra volumes on startup (only  1200MB)

`docker build -f Dockerfile --build-arg SETUP=1 -t securecompliance/gvm:v21.04-latest-withdata -t securecompliance/gvm:latest-withdata  .`

## **Braking Change:**
- Logs of each service is not pushed to the `docker logs` each service has an own logfile.

## Fixed Bug/Issues solved:
- fixes #203
- fixes #204 
- fixes #205 
- fixes #206

## Size of different builds:
```
securecompliance/gvm               data-full            eac9487a5338   8 seconds ago       4.28GB
securecompliance/gvm               data                 463c9460ea80   12 minutes ago      2.21GB
securecompliance/gvm               no-data-full         35fc2436f515   18 minutes ago      2.58GB
securecompliance/gvm               latest               d7b61ea83ba1   18 minutes ago      511MB
securecompliance/gvm               no-data              d7b61ea83ba1   18 minutes ago      511MB
```
- `latest`, `data` and `no-data` - without `--build-arg OPT_PDF=1` don't use texlive deps for the optional PDF Report creation.
- `data-full` and `no-data-full` - with `--build-arg OPT_PDF=1` use texlive deps for the optional PDF Report creation.
- `no-data` - don't run the initial setup - this will done on first start.
- `data` - have a full installation that can be used without any volumes out of the box. It has already pre-synced NVT and gvmd data.

<hr /><em>This discussion was created from the release <a href='https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/releases/tag/21.4.0-v6'>Moved Build to Alpine | Add data.tar.gx | And other hotfixes (#213)</a>.</em>

**Comment by Dexus** [comment]
No release on docker hub?
